src/components/Navigation/Pagination/PaginationPreview.tsx

```diff
@@ -0,0 +1,64 @@
+import type { FC } from "react";
+import { Trans, useTranslation } from "react-i18next";
+import { Link } from "react-router-dom";
+import { SecondaryButton, type SecondaryButtonProps } from "../../Buttons";
+import { Skeleton } from "../../Feedback";
+import { PaginationBase, type PaginationBaseProps } from "./(components)";
+import { PaginationI18nBundle as i18n } from "./Pagination.i18n";
+
+export type PaginationPreviewProps = Omit<PaginationBaseProps, "controls"> & {
+  /** A react component based on <Pagination.ViewAll /> */
+  /** If provided, the component will render as a react-router-dom Link */
+  to?: string;
+  buttonProps?: typeof SecondaryButton;
+};
+export const PaginationPreview: FC<PaginationPreviewProps> = ({
+  buttonProps,
+  to,
+  ...props
+}) => (
+  <PaginationBase
+    {...props}
+    controls={
+      <Controls
+        to={to}
+        {...buttonProps}
+        loading={props.loading}
+        count={props.count}
+      />
+    }
+  />
+);
+
+type ControlsProps = SecondaryButtonProps & {
+  count?: number;
+  loading?: boolean;
+  /** If provided, the component will render as a react-router-dom Link */
+  to?: string;
+};
+/** A specialized <Button /> with content and styling predefined */
+const Controls: FC<ControlsProps> = ({ count, loading, to, ...props }) => {
+  const { t } = useTranslation(i18n.namespace);
+
+  return (
+    <SecondaryButton
+      as={to ? Link : undefined}
+      to={to}
+      disabled={!count}
```

Comment:
When this is a Link and disabled, it can still be clicked for navigation

Reply:
Nice catch `className={twMerge(disabled && "pointer-events-none")}` helps.